repository_analysis.go

```diff
@@ -2096,12 +2102,381 @@ func (r *analysisRepository) CreateAnalysisResultQueueItem(ctx context.Context,
 	_, err = r.db.NamedExecContext(ctx, query, cerberusQueueItem)
 	if err != nil {
 		log.Error().Err(err).Msg("Failed to create cerberus queue item")
-		return uuid.Nil, nil
+		return uuid.Nil, err
 	}
 
 	return cerberusQueueItem.ID, nil
 }
 
+func (r *analysisRepository) DeleteOldCerberusQueueItems(ctx context.Context, cleanupDays, limit int) (int64, error) {
+	query := fmt.Sprintf(`DELETE FROM %s.sk_cerberus_queue_items WHERE queue_item_id IN
+				(SELECT queue_item_id FROM %s.sk_cerberus_queue_items scqi WHERE created_at <= current_date - ($1 ||' DAY')::INTERVAL LIMIT $2);`, r.dbSchema, r.dbSchema)
+	result, err := r.db.ExecContext(ctx, query, strconv.Itoa(cleanupDays), limit)
+	if err != nil {
+		log.Error().Err(err).Msg("delete old cerberus queue items failed")
+		return 0, err
+	}
+
+	return result.RowsAffected()
+}
+
+func (r *analysisRepository) DeleteOldAnalysisRequests(ctx context.Context, limit int) (int64, error) {
+	tx, err := r.getTransaction()
+	if err != nil {
+		log.Error().Err(err).Msg("delete old analysis requests failed")
+		return 0, err
+	}
+
+	txR := *r
+	txR.db = tx
+	query := fmt.Sprintf(`SELECT id FROM %s.sk_analysis_requests WHERE valid_until_time < timezone('utc', now()) AND is_processed LIMIT $1;`, r.dbSchema)
```

Comment:
would be nice to keep those which have been used, either if it has been used to sent an order msg to instrument, either if it was paired to with a result